### PR TITLE
EP-6 EP-55 Transfer send controller

### PR DESCRIPTION
**Pull request by ghiobi**
- send transfer logic
- send transfer test
- does not send token link by mail or text yet _will be implemented next commit_

### CODE REVIEW

File: tests/feature/TransferTest.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            'wallet_id' => 1,
+            'message' => 'Hello you are a friendly person'
+        ])->assertSee('errors');
+
+        $this->post('api/transfer/user/send', [
+            'receiver'  => [
+                'phone_number' => '15143338888',
+            ],
+            'amount' => 100,
+            'wallet_id' => 2, //Wrong wallet
```
Comment (Peter0892) [inline tests/feature/TransferTest.php]: Do we want to keep inline comments?

File: tests/feature/TransferTest.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            'message' => 'Hello you are a friendly person'
+        ])->assertSee('errors');
+
+        $this->post('api/transfer/user/send', [
+            'receiver'  => [
+                'phone_number' => '15143338888',
+            ],
+            'amount' => 100.001, //To many decimals
```
Comment (Peter0892) [inline tests/feature/TransferTest.php]: Do we want to keep inline comments? Please verify the same applies for all the changed files.

File: app/Http/Controllers/Transfer/User/SendController.php [hunk 3]
```diff
@@ -2,21 +2,119 @@
...
+     *
+     * @param Request $request
+     */
+    protected function validateRequest(Request $request)
+    {
+        $this->validate($request, [
+            'receiver' => 'required',
+            'receiver.phone_number' => 'required_without:receiver.email',
+            'receiver.email' => 'required_without:receiver.phone_number|email',
+            'amount' => 'numeric',
```
Comment (Peter0892) [inline app/Http/Controllers/Transfer/User/SendController.php]: Why is the amount not required?

File: tests/feature/TransferTest.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+     */
+    public function testSendTransfer()
+    {
+        $this->seed();
+
+        $user = User::find(1);
+        $this->be($user);
+
+        $this->post('api/transfer/user/send', [
+            'receiver'  => [
```
Comment (Peter0892) [inline tests/feature/TransferTest.php]: Shouldn't there be phone number inside receiver?

Comment (ghiobi) [inline tests/feature/TransferTest.php]: These are tests. That is why there should be inline comments. 
It is testing to see if there exists on receiver phone number or email field.

File: tests/feature/TransferTest.php (hunk 2 above)
Comment (ghiobi) [inline tests/feature/TransferTest.php]: Testing for too many decimals.

File: app/Http/Controllers/Transfer/User/SendController.php (hunk 3 above)
Comment (ghiobi) [inline app/Http/Controllers/Transfer/User/SendController.php]: You're right.

**Pull request merged**